lib/api/liskApi.js

```diff
@@ -163,22 +163,35 @@ LiskAPI.prototype.sendRequest = function (requestType, options, callback) {
 	return privateApi.sendRequestPromise.call(this, requestType, options).then(function (requestSuccess) {
 		var returnAnswer = (parseOfflineRequest(requestType, options).requestMethod === 'GET') ? requestSuccess.body : parseOfflineRequest(requestType, options).transactionOutputAfter(requestSuccess.body);
 
-		if (!callback || (typeof callback !== 'function')) {
-			return Promise.resolve(returnAnswer);
-		} else {
-			return callback(returnAnswer);
+		if (callback && (typeof callback === 'function')) {
```

Comment:
You call `callback` before checking for "Timestamp is in the future". The `callback` should be called only if you don't do another api call.